packages/sit-onyx/src/components/examples/GridPlayground/EditGridElementDialog/EditGridElementDialog.vue

```diff
@@ -0,0 +1,180 @@
+import OnyxCheckbox from "../../../OnyxCheckbox/OnyxCheckbox.vue";
+import OnyxDialog from "../../../OnyxDialog/OnyxDialog.vue";
+import OnyxHeadline from "../../../OnyxHeadline/OnyxHeadline.vue";
+import OnyxStepper from "../../../OnyxStepper/OnyxStepper.vue";
+
+export type GridElementConfig = {
+  columnCount: number;
+  breakpoints?: Partial<Record<OnyxBreakpoint, number>>;
+};
+
+const props = defineProps<{
+  open?: boolean;
+  /**
+   * Initial value. Useful if editing an existing element.
+   */
+  initialValue?: GridElementConfig;
+}>();
+
+const emit = defineEmits<{
+  close: [];
+  submit: [element: GridElementConfig];
+  delete: [];
+}>();
+
+const state = ref<Partial<GridElementConfig> & Required<Pick<GridElementConfig, "breakpoints">>>({
+  breakpoints: {},
+});
+
+const cloneElement = <T extends object>(value: T): T => structuredClone(toRaw(value));
+
+// reset state when dialog is closed
+watch([() => props.open, () => props.initialValue], () => {
+  if (props.initialValue) {
+    state.value = { breakpoints: {}, ...cloneElement(props.initialValue) };
+  } else {
+    state.value = { breakpoints: {} };
+  }
+});
+
+const handleSubmit = () => {
+  emit("submit", cloneElement(state.value as GridElementConfig));
+};
+
+const label = "Column configuration";
+const STEPPER_VALIDATIONS = { min: 1, max: 20 } as const;
+
+const handleCheckboxChange = (isChecked: boolean, breakpoint: OnyxBreakpoint) => {
+  if (!isChecked) {
+    delete state.value.breakpoints[breakpoint];
+  } else if (!state.value.breakpoints[breakpoint]) {
+    state.value.breakpoints[breakpoint] = state.value.columnCount ?? 1;
+  }
+};
+</script>
+
+<template>
+  <OnyxDialog :label="label" :open="props.open" modal class="dialog" @close="emit('close')">
+    <form @submit.prevent="handleSubmit">
+      <div class="dialog__header">
+        <OnyxHeadline is="h2">{{ label }}</OnyxHeadline>
+
+        <p class="dialog__description onyx-text--small">
+          Define the responsive behavior of the component by setting the number of columns.
+          Optionally, you can set different configs for each breakpoint.
+        </p>
+
+        <OnyxStepper
+          v-model="state.columnCount"
+          label="Default column count"
+          placeholder="Default column count"
+          v-bind="STEPPER_VALIDATIONS"
+          autofocus
+          required
+        />
+      </div>
+
+      <div class="dialog__body">
+        <p class="dialog__description onyx-text--small">
+          The breakpoint configs follow a "grater or equal than" logic. Example: For default column
+          4 and breakpoint "md" with 8 columns, the component will span 4 columns for breakpoints
+          smaller than md and 8 columns for md and larger.
+        </p>
+
+        <div class="dialog__grid">
+          <div
+            v-for="(_, breakpoint) in ONYX_BREAKPOINTS"
+            :key="breakpoint"
+            class="dialog__breakpoint"
+          >
+            <OnyxCheckbox
```

Comment:
for convenience, could you stretch the checkbox/label to fullwidth so you can also press on the upper right corner of the block?
![image](https://github.com/user-attachments/assets/d4b06bb6-0a43-40d6-8ff1-c134720b330b)